Update 8_1_Garden_of_Antiquity.lua

Scarface-x opened the pull request:
Updated logic and progress tracking.

**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Fixes the issue where 2 of the 3 Rubious Crystals were not spawning the NMs, thus preventing from progressing in the mission.

Issue was reported here:
https://github.com/LandSandBoat/server/issues/6018

## Steps to test these changes

Testing the mission:
I reset the quest bits to make sure I had a clean slate to test with. I then followed this guide from start to end to make sure everything progressed correctly: 

https://ffxiclopedia.fandom.com/wiki/Garden_of_Antiquity

I had no problems, NMs spawned correctly, and cutscenes were correctly displayed. Afterwards the next mission was flagged correctly. I did NOT use !pos teleporting, I ran the area as you would normally when doing the mission.

Scarface-x committed:
Update 8_1_Garden_of_Antiquity.lua

Updated logic and progress tracking.
The mission now seems to fully work from start to end.

claywar commented:
Can you please clarify the root cause that you're addressing?

Scarface-x commented:
> Can you please clarify the root cause that you're addressing?

I don't know the initial root cause of the issue, I only know the issue which was reported here is the same issue me and now 3 other players suffered: https://github.com/LandSandBoat/server/issues/6018 and apparently in that thread was reported before that, I just went through the code and did things the way I would normally, tracking and debugging each part, I have had at least 3 players get stuck on the same issue, my edit has fixed the issue as far as allowing progression where the original script did not, at least for me and 3 other players.

claywar commented:
Out of curiousity, _how_ are you defeating these mobs?  All 3 must be defeated to get credit, but if you're using GM commands, it may fail

Scarface-x commented:
> Out of curiousity, _how_ are you defeating these mobs? All 3 must be defeated to get credit, but if you're using GM commands, it may fail

Defeating then by normal attacking, I did not use any GM commands whatsoever, the issue is that 2 of the 3 nm groups will not spawn at all, I have fixed that issue, also the one nm group that did spawn, when activating the crystal again, you would not get the cutscene, i have fixed these issues, everything us in that bug report mate, I'm not the person who made the original report.

claywar commented:
So there's two different issues at play, and you didn't fix the one where the mobs aren't spawning mate.

Scarface-x commented:
> So there's two different issues at play, and you didn't fix the one where the mobs aren't spawning mate.
> 
> The issue was that the enum referenced for extended mission status was incorrect. That solves the problem for setting the correct bit since statusIndex was nil in the set/get functions. Another issue is that depending on kill method, credit may not be received in onMobDeath. This is the issue that we are trying to root cause now.

Sorry I have to disagree, 3 people have tested the mission and completed it without issue, all nms spawning, all cutscenes showing, plus myself.

Scarface-x commented:
> So there's two different issues at play, and you didn't fix the one where the mobs aren't spawning mate.
> 
> The issue was that the enum referenced for extended mission status was incorrect. That solves the problem for setting the correct bit since statusIndex was nil in the set/get functions. Another issue is that depending on kill method, credit may not be received in onMobDeath. This is the issue that we are trying to root cause now.

My edit may not be the best way to fix the issue, but it has fixed it as far as going through the mission as per the guide, try it mate.

claywar commented:
> > So there's two different issues at play, and you didn't fix the one where the mobs aren't spawning mate.
> > The issue was that the enum referenced for extended mission status was incorrect. That solves the problem for setting the correct bit since statusIndex was nil in the set/get functions. Another issue is that depending on kill method, credit may not be received in onMobDeath. This is the issue that we are trying to root cause now.
> 
> Sorry I have to disagree, 3 people have tested the mission and completed it without issue, all nms spawning, all cutscenes showing, plus myself.

Perhaps, but what you did was a bandaid that potentially broke other things.  This is why I asked for the root cause of the issue you were fixing.  Two lines were the culprit, and those have been fixed and merged.

Scarface-x commented:
> > > So there's two different issues at play, and you didn't fix the one where the mobs aren't spawning mate.
> > > The issue was that the enum referenced for extended mission status was incorrect. That solves the problem for setting the correct bit since statusIndex was nil in the set/get functions. Another issue is that depending on kill method, credit may not be received in onMobDeath. This is the issue that we are trying to root cause now.
> > 
> > 
> > Sorry I have to disagree, 3 people have tested the mission and completed it without issue, all nms spawning, all cutscenes showing, plus myself.
> 
> Perhaps, but what you did was a bandaid that potentially broke other things. This is why I asked for the root cause of the issue you were fixing. Two lines were the culprit, and those have been fixed and merged.

Maybe, but it works as far as progression, the original does not.

claywar commented:
Yep, you're absolutely correct.  There _was_ an issue with the existing code.  I'm not denying that, but our goal is to remain retail accurate as much as possible, and fix the underlying cause of issues.  This was done in #6175 as a result of investigation, but like I said, there are other scenarios that we are investigating that are unrelated to the initial issue, and are trying to gather information.

Scarface-x commented:
> Yep, you're absolutely correct. There _was_ an issue with the existing code. I'm not denying that, but our goal is to remain retail accurate as much as possible, and fix the underlying cause of issues. This was done in #6175 as a result of investigation, but like I said, there are other scenarios that we are investigating that are unrelated to the initial issue, and are trying to gather information.

If I pushed forward to getting it fixed by you fellas im happy :) Appreciate you man

Xaver-DaRed commented:
The issue refenced in this PR were fixed in PR #6175 

We do thank you for your effort. However, in this concrete case, the fix you provided was not the correct way to fix the underlying issue.

A couple of notes:
- We tend to ask questions regarding how PRs were tested or to understand the reasoning behing the fixes submited. This is not an attack, but a way to ensure quality and get to know your process.
- When submitting a PR, you have to make sure the CI checks down below pass. This checks are an automated set of tests that try to prevent bigger issues from being implemented by accident and to ensure an style across all the files.

Still, again, TY for your time and dont let this discourage you.
